src/Apps/Order/Routes/Payment/reducer.ts

```diff
@@ -0,0 +1,13 @@
+import { PaymentState, PaymentAction, PaymentActions } from "./types"
+
+export const paymentReducer = (state: PaymentState, action: PaymentAction) => {
+  switch (action.type) {
+    case PaymentActions.SET_SELECTED_BANK_ACCOUNT_ID:
+      return {
+        ...state,
+        selectedBankAccountId: action.payload,
+      }
```

Comment:
I kept the action right in the reducer, which might turn into a long long switch statement if we moved all state updaters here.

another idea is to have actions in a separate file. just focused functions. any idea?

Reply:
overall I think it'd be good to keep the reducer logic-free.

Reply:
Generally speaking, logic lives in the reducer. That way one doesn't have to go all around in order to update logic relating to state; the dev can go to one file and make changes necessary, keeping a strict separation of concerns. If things grow in complexity then further organization might be necessary, but for the most part this is all that's needed. 

Curious what your thoughts are @AlicanAkyuz?